drivers/hmis/app/jobs/hmis/ce/build_candidate_pools_job.rb

```diff
@@ -0,0 +1,38 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+#
+# Hmis::Ce::BuildCandidatePoolsJob
+#
+# Builds candidate pools for opportunities and marks them as dirty for processing.
+#
+module Hmis::Ce
+  class BuildCandidatePoolsJob < BaseJob
+    # @param opportunity_ids [Array<Integer>, nil] IDs of opportunities to process.
+    #   If nil, processes all active opportunities.
+    def perform(opportunity_ids: nil)
+      raise unless Hmis::Ce.configuration.enabled?
+
+      if opportunity_ids
+        opportunities = Hmis::Ce::Opportunity.where(id: opportunity_ids)
+      else
+        opportunities = Hmis::Ce::Opportunity.active
+      end
```

Comment:
This is tiny, and probably doesn't matter, but I've found sometimes assigning un-initialized variables within case statements can lead to unassigned variables later.  
```suggestion
      opportunities = if opportunity_ids
        Hmis::Ce::Opportunity.where(id: opportunity_ids)
      else
        Hmis::Ce::Opportunity.active
      end
```

Reply:
I'll extract this to a method to address the concern